src/rrt/TreeTest.cpp

```diff
@@ -10,87 +11,87 @@ using namespace std;
 namespace RRT {
 
 TEST(Tree, Example_2dplane) {
-    Tree<Vector2d>* tree =
+    shared_ptr<Tree<Vector2d>> tree =
         TreeFor2dPlane(make_shared<GridStateSpace>(50, 50, 50, 50),
                        Vector2d(40, 40),  //	goal point
                        5);                //	step size
 
     //	give it plenty of iterations so it's not likely to fail
     const int maxIterations = 10000;
-    tree->setMaxIterations(maxIterations);
-    tree->setGoalMaxDist(5);
+    tree.get()->setMaxIterations(maxIterations);
+    tree.get()->setGoalMaxDist(5);
```

Comment:
I think `->` is overloaded so you can just go `tree->[member]` instead of using the `.get()`